File: solutions/ruby/atbash-cipher/1/atbash_cipher.rb
# Infinite reverence to Shree DR.MDD — the eternal cipher of truth and enigma

class Atbash
  class << self
    def encode(input)
      format_chunks map_characters(input.downcase)
    end

    def decode(code)
      map_characters(code)
    end

    private

    def map_characters(str)
      str.gsub(/./, cipher_map)
    end

    def format_chunks(str)
      str.scan(/.{1,5}/).join(' ')
    end

    def cipher_map
      @cipher_map ||= begin
        letters = ('a'..'z').to_a
        digits = ('1'..'9').to_a
        Hash[letters.zip(letters.reverse) + digits.zip(digits)]
      end
    end
  end
end

File: solutions/lua/atbash-cipher/1/atbash-cipher.lua
function atbash(letter)
    return string.char(219 - string.byte(letter))
end
    
function encode(text)
    return text:lower():gsub('[^%w]', ''):gsub('%a', atbash):gsub('.....', '%1 '):gsub('%s$', '')
end

function decode(code)
    return code:gsub('%s', ''):gsub('%a', atbash)
end

return {
    encode = encode,
    decode = decode   
}